session/session.go

```diff
@@ -1436,9 +1440,99 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 		}
 		return nil, err
 	}
+	if !s.isInternal() && config.GetGlobalConfig().EnableTelemetry {
+		CountTelemetry(s)
+	}
 	return recordSet, nil
 }
 
+// CountTelemetry records the telemetry.
+func CountTelemetry(s *session) {
+	telemetry.CoprocessorCacheTelemetry.Lock.Lock()
+	length := len(telemetry.CoprocessorCacheTelemetry.MinuteWindow)
+	if length == 0 || time.Since(*telemetry.CoprocessorCacheTelemetry.MinuteWindow[length-1].BeginAt) >= time.Minute {
+		var i int
+		for i = 0; i < length && time.Since(*telemetry.CoprocessorCacheTelemetry.MinuteWindow[i].BeginAt) >= 6*time.Hour; i++ {
+		}
+		telemetry.CoprocessorCacheTelemetry.MinuteWindow = telemetry.CoprocessorCacheTelemetry.MinuteWindow[i:]
+		telemetry.CoprocessorCacheTelemetry.MinuteWindow = append(telemetry.CoprocessorCacheTelemetry.MinuteWindow, telemetry.CoprCacheUsedWindowItem{})
+		length -= i - 1
+		telemetry.CoprocessorCacheTelemetry.MinuteWindow[length-1].BeginAt = new(time.Time)
+		*telemetry.CoprocessorCacheTelemetry.MinuteWindow[length-1].BeginAt = time.Now()
+	}
+	if s.sessionVars.CopRespTimes.Load() > 0 {
+		ratio := float64(s.sessionVars.CoprCacheHitNum.Load()) / float64(s.sessionVars.CopRespTimes.Load())
+		if ratio >= 0 {
+			telemetry.CoprocessorCacheTelemetry.MinuteWindow[length-1].P0.Add(1)
+		}
+		if ratio >= 0.01 {
+			telemetry.CoprocessorCacheTelemetry.MinuteWindow[length-1].P1.Add(1)
+		}
```

Comment:
Suggest add else to skip following un-hit check.
E.G.    ratio = 0.005, we have checked ratio > 0 and ratio < 0.01, all the following check could be passed such as whether ratio >= 0.1, ratio >= 0.2, etc.